Framework/Core/include/Framework/ASoAHelpers.h

```diff
@@ -82,25 +82,44 @@ std::vector<std::pair<uint64_t, uint64_t>> doGroupTable(const T& table, const st
     selectedRows = table.getSelectedRows(); // vector<int64_t>
   }
 
-  for (uint64_t ci = 0; ci < chunkedArray->num_chunks(); ++ci) {
-    auto chunk = chunkedArray->chunk(ci);
+  auto binningColumns = binningPolicy.getColumns();
+  auto arrowColumns = o2::framework::binning_helpers::getArrowColumns(arrowTable, binningColumns);
+  auto chunksCount = arrowColumns[0]->num_chunks();
+  // TODO: Are such checks needed or can we safely assume chunks are always the same?
```

Comment:
@ktf?
If I don't need to check the alignments, then I can get rid of `getArrowColumns()` and `getChunks()` altogether.

Reply:
For sure arrow arrows mixing chunks sizes / number of chunks. I would not assume they are necessarily the same (although it might be in our case so far).